.github/actions/cdn_code_deploy/action.yaml

```diff
@@ -0,0 +1,107 @@
+name: CDN Code Deploy 
+description: Action that Deploy an artifact to the CDN
+
+# This action allows you to deploy the passed artifact to the CDN.
+# The artifact have to be a directory that contains all the files you want to sync.
+
+inputs:
+  inputs:
+    resource_group_name:
+      description: CDN resource group name.
+      type: string
+      required: true
+    storage_account_name:
+      description: Storage account name.
+      type: string
+      required: true
+    profile_cdn_name:
+      description: CDN Profile name.
+      type: string
+      required: true
+    endpoint_name:
+      description: CDN endpoint name.
+      type: string
+      required: true
+    blob_container_name:
+      description: Blob container name.
+      type: string
+      required: true
+    content_paths:
+      description: If you want to include only specific files set 'true' else 'false'
+      type: string
+      required: false
+    base_branch:
+      description: The branch to compare against the current branch. Defaults to "main" (Not necessary if 'content_paths' is 'false').
+      type: string
+      required: false
+    bundle_name:
+      description: Name of the bundle
+      type: string
+      required: false
+    arm_client_id:
+      description: ARM_CLIENT_ID
+      type: string
+      required: true
+    arm_tenant_id:
+      description: ARM_TENANT_ID
+      type: string
+      required: true
+    arm_subscription_id:
+      description: ARM_SUBSCRIPTION_ID
+      type: string
+      required: true
+runs:
+  using: composite
+  steps:
+    - name: Checkout
+      uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+    - name: Download Artifact
+      uses: actions/download-artifact@6b208ae046db98c579e8a3aa621ab581ff575935 # v4.1.1
+      with:
+        name: ${{ inputs.bundle_name || 'bundle' }}
+        path: "./${{ inputs.bundle_name || 'bundle' }}"
```

Comment:
what if i want to deploy the root directory?

Reply:
Now you can't do it directly with that action, you have to create an artifact and then download it in a next step